pkg/test/istioio/script_parser.go

```diff
@@ -0,0 +1,229 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package istioio
+
+import (
+	"strconv"
+	"strings"
+
+	"istio.io/pkg/log"
+)
+
+const (
+	snippetStartToken = "$snippet"
+	snippetEndToken   = "$endsnippet"
+
+	syntaxKey   = "syntax"
+	excludeKey  = "exclude"
+	outputisKey = "outputis"
+)
+
+var _ Step = ScriptParser{}
+
+// ScriptParser is a test Step that parses an input script which may contain a mix of raw bash commands
```

Comment:
Er, I'm a bit confused by the need for this.

The original design with snippets was that this combination was directly supported by the web site infra without any special magic, just by using comments:

```
# $snippet SNIP1
cmd1
# $endsnippet

# $snippet SNIP2
cmd3
# $endsnippet
```

So you can run this script as-is with no special parser. On the ingestion side in istio.io, the regex can pluck out the snippets in the usual way.

Is the ScriptParser giving more functionality than that?

Reply:
Ah that's a good point ... I had forgotten about the support for `# $snippet...`. So the idea would be that the entire script would be executable and we could effectively copy the content of the entire file over to the `*.snippets.txt` file.

Let me think about it some more.